osf/models/analytics.py

```diff
@@ -55,17 +55,21 @@ def increment(cls, user_id, action, date_string):
         return True
 
 
-class PageCounter(BaseModel):
-    primary_identifier_name = '_id'
+class PageCounter(BaseModel, ObjectIDMixin):
+    _id = models.CharField(max_length=50, default=generate_object_id, unique=True, db_index=True)
 
-    _id = models.CharField(max_length=300, null=False, blank=False, db_index=True,
-                           unique=True)  # 272 in prod
     date = DateTimeAwareJSONField(default=dict)
 
     total = models.PositiveIntegerField(default=0)
     unique = models.PositiveIntegerField(default=0)
 
-    DOWNLOAD_ALL_VERSIONS_ID_PATTERN = r'^download:[^:]*:{1}[^:]*$'
+    action = models.CharField(max_length=128, null=False, blank=False, default='download', db_index=True)
+    guid = models.ForeignKey(Guid, related_name='pagecounters', null=True, blank=True, db_index=True)
+    file = models.ForeignKey('osf.BaseFileNode', null=True, blank=True, related_name='pagecounters', db_index=True)
+    version = models.IntegerField(null=True, blank=True, db_index=True)
```

Comment:
I split existing _id field of form action:node_id:file_id:version into four fields.  action = charfield, version is integerfield (if null, then this is the sum of all the versions) guid=FK and file= FK.

Made most null=True and blank=True.  Do we want to enforce that they're there?